### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by wsuchy**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and open-source license: see below

#### Summary
Minor fixes like exception type, more logging ad additional args check.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

- Salesforce

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

File: cmdstanpy/model.py [hunk 1]
```diff
@@ -493,6 +493,7 @@ def _do_sample(self, stanfit: StanFit, idx: int) -> None:
         """
         cmd = stanfit.cmds[idx]
         self._logger.info('start chain {}.  '.format(idx + 1))
+        self._logger.debug("sampling: %s", cmd)
```
Comment (mitzimorris) [inline cmdstanpy/model.py]: please use new style formatting everywhere.

**Comment by maedoc** [comment]
Logging statements in principle should always pass arguments to the logging method, and not format at the call site, to avoid formatting work which isn't required.

File: cmdstanpy/model.py (hunk 1 above)
Comment (ahartikainen) [reply cmdstanpy/model.py]: Did this need some setup to work correctly (logging format style?), but I'm not sure if this affects our use of the logging module.

**Comment by wsuchy** [comment]
Indeed, see here: https://docs.python.org/3/library/logging.html#logging.Logger.debug the logger does formatting for you

File: cmdstanpy/model.py (hunk 1 above)
Comment (mitzimorris) [reply cmdstanpy/model.py]: I see, according to this:  https://docs.python.org/3/howto/logging-cookbook.html#use-of-alternative-formatting-styles  new style formatting requires config - default is old-style formatting, as above.

**Comment by bob-carpenter** [comment]
> Logging statements in principle should always pass arguments to the logging method, and not format at the call site, to avoid formatting work which isn't required.

If the logger methods are limited, you can use this convenient Java idiom:

```
if error
  if logger level > some threshold
    do some work to construct an error message
    log message at logger level
```

This is what we do before throwing exceptions in the C++ code.

**Comment by mitzimorris** [comment]
I would like to see new-style formating, e.g.  `'blah blah {}'.format(x)` used everywhere for the (foolish) sake of consistency, but it would also be OK to use old style formatting if this is what older Python 3 versions require - in which case, let's use old-style formatting with single quotes everywhere.

here's a grep for calls to `self._logger`:

`cmdstanpy/model.py` :
```
133:                self._logger.info('stan to c++ ({})'.format(hpp_file))

173:                self._logger.error('make cmd failed %s', e)

495:        self._logger.info('start chain {}.  '.format(idx + 1))
502:        self._logger.info('finish chain {}.  '.format(idx + 1))
```

 `cmdstanpy/utils.py` :
```
62:                    self._logger.debug('input tempfile: {}'.format(fd.name))
```

apologies for being a small-minded hobgoblin - do we need to explicitly adopt a style guide?
I hate to add too much process to a lightweight wrapper module.

**Comment by mitzimorris** [comment]
just read most recent comments - agreed - old style formatting, single quotes please.
see previous comments for things that need changed.

**Pull request merged**